examples/staking-broker/src/wasm.rs

```diff
@@ -0,0 +1,275 @@
+                .and_modify(|old| *old += value)
+                .or_insert(value);
+            self.total_debit += value;
+            self.has_bonded_any = true;
+            self.reward_account = match payee {
+                Some(RewardAccount::Program) | None => msg::source().into(),
+                Some(RewardAccount::Custom(account_id)) => account_id,
+            };
+        })
+        .await
+    }
+
+    async fn unbond(&mut self, value: u128) {
+        let source = msg::source();
+
+        // The sender can unbond only so much as they have bonded
+        let value = self.bonded.get(&source).map_or(0, |v| (*v).min(value));
+        if value == 0 {
+            debug!("[StakingBroker::unbond] No bonded amount");
+            msg::reply_bytes(b"No bonded amount", 0).expect("Failed to send reply");
+            return;
+        }
+
+        // Prepare a message to the built-in actor
+        let payload = Request::Unbond { value };
+        debug!(
+            "[StakingBroker] Sending `unbond` message {:?} at broker's state {:?}",
+            payload, self
+        );
+        do_send_message(payload, || {
+            // Update local state
+            if let Some(old) = self.bonded.get_mut(&source) {
+                *old = old.saturating_sub(value);
+            }
+            self.total_debit = self.total_debit.saturating_sub(value);
+        })
+        .await
+    }
+
+    async fn nominate(&mut self, targets: Vec<AccountId>) {
+        // Prepare a message to the built-in actor
+        let payload = Request::Nominate { targets };
+        debug!(
+            "[StakingBroker] Sending `nominate` message {:?} at broker's state {:?}",
+            payload, self
+        );
+        do_send_message(payload, || {}).await
+    }
+
+    async fn rebond(&mut self, value: u128) {
+        let source = msg::source();
+
+        // Prepare a message to the built-in actor
+        let payload = Request::Rebond { value };
+        debug!(
+            "[StakingBroker] Sending `rebond` message {:?} at broker's state {:?}",
+            payload, self
+        );
+        do_send_message(payload, || {
+            // Update local state
+            if let Some(old) = self.bonded.get_mut(&source) {
+                *old = old.saturating_add(value);
+            }
+            self.total_debit = self.total_debit.saturating_add(value);
+        })
+        .await
+    }
+
+    async fn witdraw_unbonded(&mut self) {
+        let _sender = msg::source();
+
+        // Prepare a message to the built-in actor
+        let payload = Request::WithdrawUnbonded {
+            num_slashing_spans: 0,
+        };
+        debug!(
+            "[StakingBroker] Sending `withdraw_unbonded` message {:?} at broker's state {:?}",
+            payload, self
+        );
+        do_send_message(payload, || {
+            // TODO: send a part of withdrawn amount to the sender and/or
+            // some other users who requested unbonding earlier
```

Comment:
It'd complicate the contract itself - we'd need to maintain more data structures in the contract to keep track of who else nominates which validator, and what are the proportions the reward should be split into so every nominator gets their share. This `TODO` is just a hint to a real contract developer that this should be taken care of.
Since it's rather urgent to get the builtin actors basics merged to not hold back other actors development, I've decided to leave this for the future.